File: teoria/clase3/clase3_Ruby/Villanos/Plan.rb
module Villanos
  class Plan

    def initialize(nombre, fecha_consecuscion, prioridad, objetivo)
      @nombre = nombre
      @fecha_consecusion = fecha_consecuscion
      @prioridad = prioridad
      @posicion_paso_siguiente = 0
      @terminado = false
      @pasos = Array.new
      @objetivo = objetivo
    end

    attr_reader :terminado

    def estaActivo
    end

    def darPasoSiguiente
    end

    def inversionPasoSiguiente
    end

  end
end

File: teoria/clase3/src/villanos/Plan.java
package villanos;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Lugli, Valentino Glauco
 */
class Plan 
{
    private ObjetivoMaligno objetivo;
    
    private String nombre;
    private Date fechaConsecucion;
    private int prioridad;
    private int posicionPasoSiguiente;
    private boolean terminado;
    
    ArrayList<Pasos> pasos;
    
    Plan(String nombre, Date fechaCon, int prioridad)
    {
        this.nombre = nombre;
        this.fechaConsecucion = fechaCon;
        this.prioridad = prioridad;
        this.posicionPasoSiguiente = 0;
        this.terminado = false;
    }
    
    public boolean getTerminado()
    {
        return this.terminado;
    }
    
    public boolean estaActivo()
    {
        throw new UnsupportedOperationException("No implementado");
    }
    
    public void darPasoSiguiente()
    {
        throw new UnsupportedOperationException("No implementado");
    }
    
    public int inversionPasoSiguiente()
    {
        throw new UnsupportedOperationException("No implementado");
    }
}
